docs/developers.md

```diff
@@ -577,19 +618,47 @@ Then, in the APB's provision tasks:
     when: proxy_passthrough
 ```
 
-# APB Spec Versioning
+## Custom Error Message
+
+A custom error message can be displayed when a failure occurs in the APB. This can be achieved by the pod writing out to its *termination log* which is by default `/dev/termination-log`.
+
+When the APB fails, the broker will pass the contents of the pod's termination log to the service catalog (if it exists), and the contents of the termination log will be displayed on the WebUI.  If the termination log is empty, a generic error message would be displayed.
+
+The below shows how this can be achieved in an APB. It captures the task in a `block` and `rescue`:
+
+```yaml
+  - block:
+      - name: Creating a DC
+        openshift_v1_deployment_config:
+          [...]
+    rescue:
+      ####################################
+      # Custom Error Message
+      ####################################
+      - name: Writing Termination Message '/dev/termination-log'
+        shell: echo "[Creating a DC Error] - {{ ansible_failed_result.msg }}" > /dev/termination-log
```

Comment:
Maybe we should create an issue for this so that we can look at what @fabianvf was talking about with callbacks?